### PR TITLE
Implemented new command !barbut

### PR DESCRIPTION
**Pull request by Malasuerte94**

### CODE REVIEW

File: cogs/events.py [hunk 1]
```diff
@@ -90,9 +89,9 @@ async def calculate_voice_time(self, member_dto):
 
     @Cog.listener()
     async def on_member_update(self, before, after):
-        dota2               = ["dota 2"]
-        bot_channel         = self.bot.get_channel(int(self.bot.channel))
-        gaming_1_channel    = self.bot.get_channel(GAMING_1_VOICE_CHANNEL)
+        dota2 = ["dota 2"]
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/events.py]: Rearrange equals one under the other

File: cogs/events.py [hunk 2]
```diff
@@ -182,28 +168,46 @@ async def on_member_join(self, member):
             await member_dto.save(self.bot)
 
     async def check_hit_and_runner(self, member_id: int):
-        config_dto      = ConfigDto()
+        config_dto = ConfigDto()
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/events.py]: Rearrange equals one under the other

File: cogs/events.py [hunk 3]
```diff
@@ -182,28 +168,46 @@ async def on_member_join(self, member):
...
         config_dto.name = config_dto.first_to_connect
         config_dto.get_config(config_dto)
 
         member_dto = MemberDto()
         member_dto.get_member(member_id)
 
         last_modified = datetime.datetime.timestamp(config_dto.modified)
-        member_left   = datetime.datetime.timestamp(member_dto.left_voice)
+        member_left = datetime.datetime.timestamp(member_dto.left_voice)
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/events.py]: Rearrange equals one under the other

File: cogs/events.py [hunk 4]
```diff
@@ -182,28 +168,46 @@ async def on_member_join(self, member):
...
 
         if member_left - last_modified < TIME_TO_VALIDATE_FLAG:
             member_dto.first_to_voice_channel = 0
             member_dto.xp += -10
             config_dto.value = 0
             config_dto.save()
-            role   = self.bot.guild.get_role(int(FIRST_TO_CONNECT_ROLE))
+            role = self.bot.guild.get_role(int(FIRST_TO_CONNECT_ROLE))
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/events.py:187]: Rearrange equals one under the other

File: cogs/events.py [hunk 5]
```diff
@@ -182,28 +168,46 @@ async def on_member_join(self, member):
...
             member = self.bot.guild.get_member(int(member_dto.member_id))
             await member.remove_roles(role)
             await member_dto.save(self.bot)
             return True
         return False
 
+    async def check_kick_command(self, embed, message, payload):
+        description = embed.description.split(':')
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/events.py]: Rearrange equals one under the other

File: cogs/games.py [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+class Games(Cog):
+    def __init__(self, bot):
+        self.bot = bot
+
+    @Cog.listener()
+    async def on_ready(self):
+        if not self.bot.ready:
+            self.bot.cogs_ready.ready_up(os.path.basename(__file__))
+
+    @command(name='barbut', pass_context=True)
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/games.py]: Add description to command: description='Description'

File: cogs/games.py [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+        for reaction in reactions:
+            await react.add_reaction(reaction)
+
+        def check(reaction, user):
+            return user == member and str(reaction.emoji) == '✅'
+
+        try:
+            await self.bot.wait_for('reaction_add', timeout=30.0, check=check)
+        except asyncio.TimeoutError:
+
```
Comment (AdrianMitroiuGeorgescu) [inline cogs/games.py]: Delete this empty line. There is no need for empty lines before first line of the code block

File: services/eventsServices.py [hunk 8]
```diff
@@ -0,0 +1,108 @@
+import asyncio
+import random
+
+from entities.members import MemberDto
+
+
+class EventsServices:
+
+    async def check_barbut_command(self, bot, embed, message, payload):
+        description = embed.description.split(':')
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: Rearrange equals one under the other

File: services/eventsServices.py [hunk 9]
```diff
@@ -0,0 +1,108 @@
...
+
+        to_roll_player = bot.guild.get_member(int(to_roll_player_id))
+
+        if payload.emoji.name == '✅' and payload.member.id == player_two:
+            await message.clear_reactions()
+            reactions = ['🎲']
+            for reaction in reactions:
+                await message.add_reaction(reaction)
+
+            # timer
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: No need for this comment

File: services/eventsServices.py [hunk 10]
```diff
@@ -0,0 +1,108 @@
...
+            def check(reaction, user):
+                return user in [payload.member, to_roll_player] and str(reaction.emoji) == '🎲'
+
+            try:
+                await bot.wait_for('reaction_add', timeout=30.0, check=check)
+            except asyncio.TimeoutError:
+                embed.add_field(name='Jocul a expirat', value='Nimeni nu a dat cu zarul', inline=True)
+                await message.edit(embed=embed)
+                await message.clear_reactions()
+            # timer
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: No need for this comment

File: services/eventsServices.py [hunk 11]
```diff
@@ -0,0 +1,108 @@
...
+        if embed.fields:
+            for field in embed.fields:
+                allready_rolled.append(field.name)
+                player_name = field.name
+                rolled_first_value = int(field.value)
+
+        if payload.emoji.name == '🎲' and payload.member.id in [player_one,
+                                                                player_two] and payload.member.display_name not in allready_rolled:
+
+            discord_member = bot.guild.get_member(payload.member.id)
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: Cast as int to minimize errors int(payload.member.id)

File: services/eventsServices.py [hunk 12]
```diff
@@ -0,0 +1,108 @@
...
+
+            roll = random.randrange(1, 100)
+            embed.add_field(name=f'{discord_member.display_name}', value=roll, inline=True)
+            await message.edit(embed=embed)
+
+            if 1 <= len(allready_rolled):
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: This variable will never be more than 1. Better check simply if lenght is more then 0

File: services/eventsServices.py [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+                    await self.games_rewarding(bot, to_roll_player_id, payload.member.id, stake_is)
+                elif rolled_first_value < roll:
+                    embed.add_field(name=f':crown: Câștigătorul este {payload.member.display_name}',
+                                    value=f'A câștigat {stake_is} XP', inline=False)
+                    await self.games_rewarding(bot, payload.member.id, to_roll_player_id, stake_is)
+                elif rolled_first_value == roll:
+                    embed.add_field(name=':crown: Egalitate', value='Nimeni nu a câștigat', inline=False)
+
+            else:
+
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py]: No need for this empty line

File: services/eventsServices.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+                    embed.add_field(name=f'{to_roll_player.display_name}', value='Predat', inline=True)
+                    embed.add_field(name=f':crown: Câștigătorul este {payload.member.display_name}',
+                                    value=f'A câștigat {stake_is} XP', inline=False)
+
+                    await self.games_rewarding(bot, payload.member.id, to_roll_player.id, stake_is)
+
+            embed.set_footer(text="")
+            await message.edit(embed=embed)
+            await message.clear_reactions()
+
```
Comment (AdrianMitroiuGeorgescu) [inline services/eventsServices.py:92]: No need for this empty line

**Review by AdrianMitroiuGeorgescu — CHANGES_REQUESTED** [review]
Solved all comments before merge

File: cogs/games.py (hunk 6 above)
Comment (Malasuerte94) [reply cogs/games.py]: Ce e asta? ca nu inteleg

Comment (AdrianMitroiuGeorgescu) [reply cogs/games.py]: Resolved on discord

**Review by AdrianMitroiuGeorgescu — DISMISSED** [review]

**Review by AdrianMitroiuGeorgescu — APPROVED** [review]

**Pull request merged**